### DISCUSSION TITLE
not able to "Enable groups"

### DISCUSSION THREAD

**Discussion by WoutAndAbout** [dani-garcia/vaultwarden#5284] [Q&A]
### Your environment (Generated via diagnostics page)
* Vaultwarden version: v1.32.5
* Web-vault version: v2024.6.2c
* OS/Arch: linux/x86_64
* Running within a container: true (Base: Debian)
* Environment settings overridden: false
* Uses a reverse proxy: true
* IP Header check: true (X-Real-IP)
* Internet access: true
* Internet access via a proxy: false
* DNS Check: true
* Browser/Server Time Check: true
* Server/NTP Time Check: true
* Domain Configuration Check: true
* HTTPS Check: true
* Database type: SQLite
* Database version: 3.46.0
* Clients used: 
* Reverse proxy and version: 
* Other relevant information: 

### Config (Generated via diagnostics page)
<details><summary>Show Running Config</summary>

**Environment settings which are overridden:** 

```json
{
  "_duo_akey": null,
  "_enable_duo": true,
  "_enable_email_2fa": true,
  "_enable_smtp": true,
  "_enable_yubico": true,
  "_icon_service_csp": "",
  "_icon_service_url": "",
  "_ip_header_enabled": true,
  "_max_note_size": 10000,
  "_smtp_img_src": "cid:",
  "admin_ratelimit_max_burst": 3,
  "admin_ratelimit_seconds": 300,
  "admin_session_lifetime": 20,
  "admin_token": "***",
  "allowed_iframe_ancestors": "",
  "attachments_folder": "data/attachments",
  "auth_request_purge_schedule": "30 * * * * *",
  "authenticator_disable_time_drift": false,
  "data_folder": "data",
  "database_conn_init": "",
  "database_max_conns": 10,
  "database_timeout": 30,
  "database_url": "***************",
  "db_connection_retries": 15,
  "disable_2fa_remember": false,
  "disable_admin_token": false,
  "disable_icon_download": false,
  "domain": "*****://*******************",
  "domain_origin": "*****://*******************",
  "domain_path": "",
  "domain_set": true,
  "duo_context_purge_schedule": "30 * * * * *",
  "duo_host": null,
  "duo_ikey": null,
  "duo_skey": null,
  "duo_use_iframe": false,
  "email_2fa_auto_fallback": false,
  "email_2fa_enforce_on_verified_invite": false,
  "email_attempts_limit": 3,
  "email_change_allowed": true,
  "email_expiration_time": 600,
  "email_token_size": 6,
  "emergency_access_allowed": false,
  "emergency_notification_reminder_schedule": "0 3 * * * *",
  "emergency_request_timeout_schedule": "0 7 * * * *",
  "enable_db_wal": true,
  "enable_websocket": true,
  "enforce_single_org_with_reset_pw_policy": false,
  "event_cleanup_schedule": "0 10 0 * * *",
  "events_days_retain": null,
  "experimental_client_feature_flags": "fido2-vault-credentials",
  "extended_logging": true,
  "helo_name": null,
  "hibp_api_key": null,
  "http_request_block_non_global_ips": true,
  "http_request_block_regex": null,
  "icon_blacklist_non_global_ips": true,
  "icon_blacklist_regex": null,
  "icon_cache_folder": "data/icon_cache",
  "icon_cache_negttl": 259200,
  "icon_cache_ttl": 2592000,
  "icon_download_timeout": 10,
  "icon_redirect_code": 302,
  "icon_service": "internal",
  "incomplete_2fa_schedule": "30 * * * * *",
  "incomplete_2fa_time_limit": 3,
  "increase_note_size_limit": false,
  "invitation_expiration_hours": 120,
  "invitation_org_name": "Vault",
  "invitations_allowed": true,
  "ip_header": "X-Real-IP",
  "job_poll_interval_ms": 30000,
  "log_file": null,
  "log_level": "info",
  "log_timestamp_format": "%Y-%m-%d %H:%M:%S.%3f",
  "login_ratelimit_max_burst": 10,
  "login_ratelimit_seconds": 60,
  "org_attachment_limit": null,
  "org_creation_users": "",
  "org_events_enabled": false,
  "org_groups_enabled": false,
  "password_hints_allowed": true,
  "password_iterations": 600000,
  "push_enabled": false,
  "push_identity_uri": "https://identity.bitwarden.com",
  "push_installation_id": "***",
  "push_installation_key": "***",
  "push_relay_uri": "https://push.bitwarden.com",
  "reload_templates": false,
  "require_device_email": false,
  "rsa_key_filename": "data/rsa_key",
  "send_purge_schedule": "0 5 * * * *",
  "sendmail_command": null,
  "sends_allowed": true,
  "sends_folder": "data/sends",
  "show_password_hint": false,
  "signups_allowed": true,
  "signups_domains_whitelist": "************,******************,*************,**************",
  "signups_verify": false,
  "signups_verify_resend_limit": 6,
  "signups_verify_resend_time": 3600,
  "smtp_accept_invalid_certs": false,
  "smtp_accept_invalid_hostnames": false,
  "smtp_auth_mechanism": null,
  "smtp_debug": false,
  "smtp_embed_images": true,
  "smtp_explicit_tls": null,
  "smtp_from": "************************",
  "smtp_from_name": " Vault",
  "smtp_host": "************",
  "smtp_password": "***",
  "smtp_port": 587,
  "smtp_security": "starttls",
  "smtp_ssl": null,
  "smtp_timeout": 15,
  "smtp_username": "************************",
  "templates_folder": "data/templates",
  "tmp_folder": "data/tmp",
  "trash_auto_delete_days": null,
  "trash_purge_schedule": "0 5 0 * * *",
  "use_sendmail": false,
  "use_syslog": false,
  "user_attachment_limit": null,
  "user_send_limit": null,
  "web_vault_enabled": true,
  "web_vault_folder": "web-vault/",
  "yubico_client_id": null,
  "yubico_secret_key": null,
  "yubico_server": null
}
```
</details>

### Vaultwarden Build Version

v1.32.5

### Deployment method

Official Container Image

### Custom deployment method

Via docker compose

### Reverse Proxy

npm 2.12.1

### Host/Server Operating System

Linux

### Operating System Version

Ubuntu 22.04.2 LTS

### Clients

Web Vault, CLI

### Client Version

Ubuntu 22.04.2 LTS ssh bash

### Steps To Reproduce

Just trying to enable the groups function.

Via .env file
Via docker-compose.yaml file
Via /admin
Via /data/config.json

None of these changes made a change to the fact that "Enable groups (BETA!) (Know the risks!)" is still disabled

### Expected Result

To be able to enable "Enable groups (BETA!) (Know the risks!)"

### Actual Result

It is not enabling the 'Enable groups (BETA!) (Know the risks!)" feature

### Logs

```text
when i tried with the config.json it told me that i used the config.json.

Then it says what it uses from the config.json. "Enable groups (BETA!) (Know the risks!)" wasn't one of them
```

### Additional Context

Let me know if the information i gave is sufficient for troubleshooting

**Comment by stefan0xC** [comment]
> when i tried with the config.json it told me that i used the config.json.

Well, the option is in the Read only section, so you can't change it via the `/admin` interface:
![image](https://github.com/user-attachments/assets/a5e30393-1820-4a4a-a6c7-b7553218ee7f)

Furthermore because it's deemed a risky operation you wouldn't see if it were overwritten:
![image](https://github.com/user-attachments/assets/aba89319-75dd-49a1-bb68-161792ea17cd)
Enabled via .env file (`ORG_GROUPS_ENABLED=true`) but I've edited the `config.json` manually (`"org_groups_enabled": false,`).

And while technically possible I would not recommend changing it manually via `config.json` because if I click save again in `/admin` panel my changes to the options in the read only section will be lost (as [noted in the wiki](https://github.com/dani-garcia/vaultwarden/wiki/Configuration-overview#using-the-admin-page)).

I've no idea why setting the environment variable did not take in your case but you might not have recreated the container?

**Reply by eduardojonck** [reply]
Good afternoon,

I understand that the information is in BETA format, but I wanted to see if anyone is using it and if they are having any difficulties. I am leaving TeamPass and there is this group option there. What do you think?
